### DISCUSSION TITLE
Feature request: Logger

### DISCUSSION THREAD

**Discussion by MickL** [nitrojs/nitro#2792] [Ideas]
Every "big" framework provide a logger (e.g. [NestJS](https://docs.nestjs.com/techniques/logger) and [AdonisJS](https://docs.adonisjs.com/guides/digging-deeper/logger).

To me personally I would be happy to be able to log to console, e.g. with [consola](https://github.com/unjs/consola), but I know a lot of people also need to log to files, or maybe use external logging services.

Needed features
* Set log level (in nitro.config.ts?) depending on environment, e.g. verbose on dev, only warn/error on prod
* Provide a context, e.g. in route /checkout it could be `const logger = useLogger('checkout')` and then the context is always prefixed

**Comment by MickL** [comment]
What makes Nitro so powerful is that on one hand it is very flexible but on the other there is already the huge UnJS ecosystem. Therefor this feature request can already be solved by adding a simple example to the docs how to create a logger, e.g.:

`bun add consola`

```ts
import { createConsola } from 'consola';
import type { ConsolaInstance } from 'consola/core';

let consola: ConsolaInstance;

export const useLogger = () => {
  if (!consola) {
    const { logLevel } = useRuntimeConfig();

    consola = createConsola({
      level: logLevel,
    });
  }

  return consola;
};
```

**Comment by cjpearson** [comment]
Having `event.$log` as suggested here would also be nice. https://github.com/unjs/nitro/issues/651#issuecomment-1425546335
